doc/doxygen/src/board-cpu-features.md

```diff
@@ -0,0 +1,99 @@
+
+FEATURES                                                             {#features}
+========
+
+`FEATURES_PROVIDED` are hardware (including toolchain) features (e.g.:`periph_hwrng`,
+`periph_uart`) that are available or characteristics (e.g:`arch_32bits`) of a board.
+
+Providing a `FEATURE`
+--------------------
+
+For a `FEATURE` to be provided by a `board` it must meet 2 criteria, and for
+`periph` and other `hw` it must follow the 3rd criteria.
+
+- Needs the "hardware" or BSP support (toolchain)
+    - e.g.: `stm32l152re` has an SPI peripheral
+            `riotboot` needs to be able to link and flash at an offset
+- Needs support in RIOT, an implementation of an api to interact with the `hw`
+    - e.g.: `cpu/stm32_common/periph/spi.c` is implemented for `stm32l1`
+            `riotboot` needs an implementation of `cpu_jump_to_image`
+- Wiring between the `cpu/soc` a `bus` and other `cpu`/`hw` components.
+    - e.g.: `nucleo-l152re/include/periph_conf.h` specified wiring between `PORT_Ax`
+             and `SPI1`
+
+What is a `FEATURE`
+--------------------
+
+Whenever a `FEATURE` is used there should be at some level a HW requirements,
+whether this is a `radio`, a `bus` ot a specific core architecture. 
+
+This is not a hard line, in some cases the line can be harder to establish than
+others. There are complicated cases like `netif` since a network interface could
+be fully implemented in software as a loop-back.
+
+It's also important to note that a `FEATURE` does not mean there is a `MODULE`
+with the same name. There could be many implementations for the same `FEATURE`.
+The fact that many `FEATURES` translate directly into a `MODULE` is only by
+convenience.
+
+e.g.
+
+    # all periph features correspond to a periph submodule
+    USEMODULE += $(filter periph_%,$(FEATURES_USED))
+
+A `FEATURE` is a tool to specify valid/invalid dependencies and configurations.
+
+
+CPU/CPU_MODEL                                                             {#cpu}
+======
+
+`CPU` and `CPU_MODEL` refer to the `SOC` (system on a chip) or `MCU` (microcontroller)
+present in a `BOARD`. The variables `CPU`, `CPU_FAM`, etc. are just arbitrary groupings
+to avoid code duplication. How this grouping is done depends on every implementation
+and the way each manufacturer groups there products.
+
+These variables allows declaring the `FEATURES` that the mcu/soc provides as well
+resolving dependencies.
+
+`FEATURES` provided by a `CPU/CPU_MODEL` should not depend on the wiring of a
+specific `BOARD` but be intrinsic to the soc/mcu.
+
+A `CPU/CPU_MODEL` might support `FEATURES` that will depend the `BOARD` wiring,
+e.g.: bus (`uart`, `spi`) mappings. In this cases the `FEATURE` should be provided
+by the `BOARD.`
```

Comment:
Good to precise this one. It is a tricky one to get by default and really shows that the 3 criteria must be there. 
The silicon might have it, there may be common code, but not be wired on the board or just not yet described in a `periph_conf`.